tilavarauspalvelu/api/graphql/types/application/serializers.py

```diff
@@ -105,6 +103,227 @@ def validate(self, data: dict[str, Any]) -> dict[str, Any]:
+        if self.instance.billing_address is None:
+            msg = "Application billing address is required."
+            errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+        else:
+            if not self.instance.billing_address.street_address:
+                msg = "Application billing address must have a street address."
+                errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+            if not self.instance.billing_address.post_code:
+                msg = "Application billing address must have a post code."
+                errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+            if not self.instance.billing_address.city:
+                msg = "Application billing address must have a city."
+                errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+    def validate_contact_person(self, errors: defaultdict[str, list[str]]) -> None:
+        if self.instance.contact_person is None:
+            msg = "Application contact person is required."
+            errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+        else:
+            if not self.instance.contact_person.first_name:
+                msg = "Application contact person must have a first name."
+                errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+            if not self.instance.contact_person.last_name:
+                msg = "Application contact person must have a last name."
+                errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+            if not self.instance.contact_person.email:
+                msg = "Application contact person must have an email address."
+                errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+            if not self.instance.contact_person.phone_number:
+                msg = "Application contact person must have a phone number."
+                errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+    def validate_organisation(
+        self,
+        errors: defaultdict[str, list[str]],
+        *,
+        allowed_organisation_types: Collection[OrganizationTypeChoice],
+        require_home_city: bool = False,
+        require_identifier: bool = False,
+    ) -> None:
+        organisation = self.instance.organisation
+
+        if organisation is None:
+            msg = "Application organisation is required."
+            errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+        else:
+            if not organisation.name:
+                msg = "Application organisation must have a name."
+                errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+            if not organisation.core_business:
+                msg = "Application organisation must have a core business."
+                errors[api_settings.NON_FIELD_ERRORS_KEY].append(msg)
+
+            if OrganizationTypeChoice(organisation.organisation_type) not in allowed_organisation_types:
```

Comment:
organisation_type is no longer saved for applications, so can't be validated
e.g. https://tilavaraus.test.hel.ninja/admin/tilavarauspalvelu/application/1053/change/ is an application for `COMMUNITY` but its org.type is the default of `COMPANY`